server/entity/experience.go

```diff
@@ -0,0 +1,153 @@
+package entity
+
+import (
+	"fmt"
+	"math"
+	"sync"
+)
+
+// ExperienceManager manages experience and levels for entities, and provides functions to add, remove, and calculate
+// experience needed for upcoming levels.
+type ExperienceManager struct {
+	mu         sync.RWMutex
+	total      int
+	experience int
```

Comment:
Do you reckon we could actually only have the `total` field here and calculate the per-level `experience` on the fly where necessary?

Reply:
Pretty sure that total is for lifetime experience